### ISSUE TITLE
Spread scheduler not spreading evenly

### ISSUE THREAD

**Issue by AxelVoitier** [docker/swarm#671]
Hello,

I am seeing a situation where the spread scheduler seems to not evenly distribute the containers over all nodes.
Well, I guess it is spread, as I did not set it on the swarm manage command line and according to doc the default is spread.

I have this state as an initial and steady state:
```
$ docker info
Containers: 8
Nodes: 4
 docker02: ***.***.***.***:2376
  └ Containers: 1
  └ Reserved CPUs: 0 / 4
  └ Reserved Memory: 0 B / 7.798 GiB
 docker04: ***.***.***.***:2376
  └ Containers: 1
  └ Reserved CPUs: 0 / 4
  └ Reserved Memory: 0 B / 7.798 GiB
 docker03: ***.***.***.***:2376
  └ Containers: 1
  └ Reserved CPUs: 0 / 4
  └ Reserved Memory: 0 B / 7.798 GiB
 docker01: ***.***.***.***:2376
  └ Containers: 5
  └ Reserved CPUs: 0 / 4
  └ Reserved Memory: 0 B / 7.798 GiB
```

The fact that the first node, docker01, has more containers running is normal, as it is the "master" node running other steady services not doing much on the cpu. I pinned them on this node (manually starting them directly on it).

Now I want to spawn 16 containers over the swarm. These containers will be short lived (10-15 seconds). I start them as fast as my script can, pretty much sequentially (of course, not waiting for the containers to stop before starting the next one).

(If that seems odd to you the use case is to spawn mono-threaded data analyses processes on a "docker cloud". And the rational behind spawning 16 of them is that I have 4 machines with 4 cores each, giving me pretty much 16 cores at disposal (neglecting the small services running on docker01 node)).

Each container is spawn with such command:
```
docker run --rm localhost:5000/my_image python my_script.py some args
```

Here is the status of the swarm in the middle of the run:
```
$ docker info
Containers: 24
Nodes: 4
 docker02: ***.***.***.***:2376
  └ Containers: 11
  └ Reserved CPUs: 0 / 4
  └ Reserved Memory: 0 B / 7.798 GiB
 docker04: ***.***.***.***:2376
  └ Containers: 1
  └ Reserved CPUs: 0 / 4
  └ Reserved Memory: 0 B / 7.798 GiB
 docker03: ***.***.***.***:2376
  └ Containers: 3
  └ Reserved CPUs: 0 / 4
  └ Reserved Memory: 0 B / 7.798 GiB
 docker01: ***.***.***.***:2376
  └ Containers: 9
  └ Reserved CPUs: 0 / 4
  └ Reserved Memory: 0 B / 7.798 GiB
```

Basically, node docker02 receives 10 containers, docker01 receives 4 containers, docker03 receives 2 containers and docker04 receives none.
Which is not an even spread at all I would say... I have one node, docker02, busted with jobs, and another, docker04, doing nothing.
I could have expected the spread to be slightly unbalanced because of the containers already running in docker01. But here it is a very non-intuitive spread (and not efficient at all).

Is there something I am missing to get the most out of the swarm here?
For info, the swarm itself was deployed by docker machine. That was a couple of weeks ago so I guess it is most likely version 0.1 of swarm.

Would upgrading to 0.2 improve the distributions of containers?
(usually I test with latest released versions before reporting any problem, but upgrading the trio docker-machine-swarm to their latest version might take me a bit of time, and as I do this only out of evaluation purpose I prefer spending time on upgrading it only if it brings sufficient added value for my evaluation)

Here is the swarm manage command line arguments:
```
"Args": [
        "manage",
        "--tlsverify",
        "--tlscacert=/etc/docker/ca.pem",
        "--tlscert=/etc/docker/server.pem",
        "--tlskey=/etc/docker/server-key.pem",
        "-H",
        "tcp://0.0.0.0:3376",
        "token://**********"
    ],
```

**Comment by abronan** [comment]
Hi @AxelVoitier, Swarm `v0.1.0` only has `random` and `bin-packing` for strategies which explains the fact that your containers are not evenly distributed over Swarm agents. `Spread` was introduced as a default strategy in `v0.2.0`. Try again with the new version and tell us if you encounter any issue ;)

**Comment by aluzzardi** [comment]
@AxelVoitier Could you provide the output of `docker version`?

**Comment by abronan** [comment]
I will close this for now, @AxelVoitier please feel free to comment here if you still have an issue, but I believe that you are using swarm `v0.1.0` :)

**Comment by AxelVoitier** [comment]
Hello,

Sorry for late answer. I was carried away by other work.
I confirm at the time I opened this issue I was using version v0.1.0.

Today, I redeployed the swarm on a new set of virtual machines (got 24 cores to play with now :)). So I took the opportunity to use v0.2.0.
After a basic test I confirm it is now spreading as expected, and that is very cool :sunglasses:.

Thanks for your good work guys!
